### PR TITLE
Chapter 3: Mostly Static Pages

### PR DESCRIPTION
**Pull request by Arizar98**

### CODE REVIEW

**Comment by Arizar98** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  margin-bottom: 10px;
+}
+
+/* header */
+
+#logo {
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: #fff;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: image

File: app/assets/stylesheets/static_pages.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/static_pages.scss:3]: remove file that never used

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
+
+  def contact
+  end
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]: image

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+  end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
    base_title = "Ruby on Rails Tutorial Sample App"
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
  end
```

File: app/helpers/static_pages_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
+end
```
Comment (saobv-2223) [inline app/helpers/static_pages_helper.rb:2]: remove file that never used

File: app/mailers/application_mailer.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
-  layout 'mailer'
+
+  default from: "from@example.com"
+  layout "mailer"
+
```
Comment (saobv-2223) [inline app/mailers/application_mailer.rb]: remove empty line

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+      <%= link_to "sample app", "#", id: "logo" %>
+      <nav>
+        <ul class="nav navbar-nav navbar-right">
+          <li><%= link_to "Home", "#" %></li>
+          <li><%= link_to "Help", "#" %></li>
+          <li><%= link_to "Log in", "#" %></li>
+        </ul>
+      </nav>
+    </div>
+  </header>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: indenting only 2 spaces

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title(yield(:title)) %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: remove ()

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+<h1>Help</h1>
+
+<p>
+  Get help on the Ruby on Rails Tutorial at the
+  <a href="http://www.railstutorial.org/help">Rails Tutorial
+  help page</a>.
+  To get help on this sample app, see the
+  <a href="http://www.railstutorial.org/book"><em>Ruby on Rails
+  Tutorial</em>
+  book</a>.
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: using link_to instead of a tag

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+  This is the home page for the
+  <a href="http://www.railstutorial.org/">Ruby on
+  Rails Tutorial</a>
+  sample application.
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: using link_to instead of a tag

File: config/database.yml [hunk 11]
```diff
@@ -1,25 +1,16 @@
-# SQLite. Versions 3.8.0 and up are supported.
-#   gem install sqlite3
-#
-#   Ensure the SQLite 3 gem is defined in your Gemfile
-#   gem 'sqlite3'
-#
+
```
Comment (saobv-2223) [inline config/database.yml:1]: image

File: config/routes.rb [hunk 12]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'static_pages#home'
```
Comment (saobv-2223) [inline config/routes.rb]: image

**Review by saobv-2223 — COMMENTED** [review]
Your PR is having 2 commits, merge them please

**Comment by Arizar98** [comment]
Ready

**Comment by ngocvt-0484** [comment]
please change 5 commits to 1 commit
you can run:
git reset --soft origin/master
git commit -m "chapter 3"
git push origin chapter_3 -f

**Comment by manhnt-0960** [comment]
please fix 5 commits

**Comment by manhnt-0960** [comment]
commented

**Comment by Arizar98** [comment]
Ready

File: app/helpers/application_helper.rb [hunk 13]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: try to learn about i18n
and use it here
https://github.com/ruby-i18n/i18n

**Comment by Arizar98** [comment]
Ready

**Review by saobv-2223 — COMMENTED** [review]
your pr is having 2 commits, merge to 1 commit, please

**Comment by Arizar98** [comment]
Ready

**Comment by chungpv-1008** [comment]
Please merge 2 commits to 1 commit

**Comment by Arizar98** [comment]
Ready

**Review by chungpv-1008 — COMMENTED** [review]
commented

**Review by chungpv-1008 — COMMENTED** [review]
Please merge 2 commits to 1 commit

File: .gitignore [hunk 14]
```diff
@@ -39,7 +39,7 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
-/config/credentials.yml.enc.example
+/config/credentials.yml.enc
```
Comment (chungpv-1008) [inline .gitignore:42]: Please merge 2 commits to 1 commit

Comment (chungpv-1008) [inline .gitignore:42]: Please merge 2 commits to 1 commit

Comment (LVQA1998) [reply .gitignore:42]: Please merge 2 commits to 1 commit

**Comment by Arizar98** [comment]
Ready

File: app/assets/stylesheets/static_pages.scss [hunk 15]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (chungpv-1008) [inline app/assets/stylesheets/static_pages.scss:1]: please remove this file because not use

File: config/database.yml [hunk 16]
```diff
@@ -1,25 +1,16 @@
...
 default: &default
-  adapter: sqlite3
-  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  timeout: 5000
+  adapter: mysql2
```
Comment (chungpv-1008) [inline config/database.yml:3]: Please ignore this file

**Comment by vanvtt-0952** [comment]
merged
Please fix comment, then merge the pull request yourself

**Pull request merged**